app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -62,10 +66,18 @@ protected function validator(array $data)
      */
     protected function create(array $data)
     {
-        return User::create([
-            'name' => $data['name'],
+        $user = User::create([
+            'username' => $data['username'],
             'email' => $data['email'],
             'password' => bcrypt($data['password']),
+            'user_type' => (($data['user_type'] == 'author') ? 2 : 3)
         ]);
+
+        User::find($user['id'])->userDetail()->save(new UserDetail([
```

Comment:
This will work, no need of refetching user
```
$user->userDetail()->save(new UserDetail([
     'firstname' => $data['firstname'],
     'lastname' => $data['lastname']
]));
```